.gitlab-ci.yml

```diff
@@ -210,42 +240,62 @@ check:format:
   stage: check
   needs: []
 
+check:bundlewatch:
+  <<: *tmpl_branch
+  script:
+    - export CI_REPO_OWNER="$CI_PROJECT_NAMESPACE"
+    - export CI_REPO_NAME="$CI_PROJECT_NAME"
+    - export CI_BRANCH="$CI_COMMIT_BRANCH"
+    - yarn bundlewatch
+  stage: check
+  needs:
+    - build:ts:branch
+  dependencies:
+    - build:ts:branch
+
 ####################
 # stage deploy-dev #
 ####################
-deploy:sls:branch:
+
+.deploy:sls:
   <<: *tmpl_branch
   image: $INTEGRATION_DOCKER_IMAGE
   stage: deploy-dev
+  script:
+    - source ci/env-setup.sh
+    - yarn sls deploy --package .serverless --verbose
+
+deploy:sls:branch:
+  extends: .deploy:sls
+  rules:
+    - if: $CI_EXTERNAL_PULL_REQUEST_IID
   variables:
     <<: *tmpl_branch_variables
     <<: *tmpl_deploy_variables
-  script:
+  before_script:
     - export SQUIDEX_APP_NAME=${CI_EXTERNAL_PULL_REQUEST_IID}
     - export SQUIDEX_CLIENT_ID=$SQUIDEX_TEST_CLIENT_ID
     - export SQUIDEX_CLIENT_SECRET=$SQUIDEX_TEST_CLIENT_SECRET
-    - source ci/env-setup.sh
-    - yarn sls deploy --package .serverless --verbose
   dependencies:
```

Comment:
i think this has been here all along (previously the branch template would also specify these are dependencies) - but what if we start deploying to PR before finishing all tests? If this might optimise the pipeline i guess we could try that, unless the tests are not really on critical path anyway in which case it makes sense to keep it simple and wait for the entire stage to finish.

Reply:
Dependencies are different to needs. Dependecies say that we want to consume an artifact. The ommitted needs is what says all previous stages are required to be successful

Reply:
i am well aware, which is why i said if we should use needs instead and have it wait for just the selected jobs since we dont need all of them